projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/trait-wpcom-rest-api-proxy-request-trait.php

```diff
@@ -8,21 +8,24 @@
  */
 
 use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager;
 use Automattic\Jetpack\Status\Visitor;
 
 trait WPCOM_REST_API_Proxy_Request_Trait {
 
 	/**
-	 * Proxy request to wpcom servers for the site and user.
+	 * Proxy request to wpcom servers on behalf of a user or using the Site-level Connection (blog token).
 	 *
 	 * @param WP_Rest_Request $request Request to proxy.
 	 * @param string          $path Path to append to the rest base.
+	 * @param string          $context Whether the request should be proxied on behalf of the current user or using the Site-lecl Connection, aka 'blog' token. Can be Either 'user' or 'blog'. Defaults to 'user'.
+	 * @param bool            $allow_fallback_to_blog If the $context is 'user', whether we should fallback to using the Site-level Connection in case the current user is not connected.
 	 *
 	 * @return mixed|WP_Error           Response from wpcom servers or an error.
 	 */
-	public function proxy_request_to_wpcom_as_user( $request, $path = '' ) {
+	public function proxy_request_to_wpcom( $request, $path = '', $context = 'user', $allow_fallback_to_blog = false ) {
```

Comment:
I was wondering if `blog` should be the default instead. It feels like we should encourage folks to make blog requests whenever possible, and user requests only when necessary.

I really like the idea of the fallback whenever possible though, so we can still keep that I think.

Reply:
That makes sense, true. 👍